ext/digest/crc.h

```diff
@@ -0,0 +1,70 @@
+#ifndef RUBY_DIGEST_CRC_H
+#define RUBY_DIGEST_CRC_H 1
+
+#include <ruby.h>
+#include <stddef.h>
+#include <stdint.h>
+
+enum crc_input_direction {
+	crc_normal_input  = 0,
+	crc_reflect_input = 1
+};
+
+enum crc_allocation_type {
+	crc_alloc_static  = 0,
+	crc_alloc_heap    = 1
+};
+
+struct crc_model
+{
+	int                      bitwidth      : 8;
+	enum crc_input_direction reflect_input : 1;
+	enum crc_allocation_type table_heap    : 1;
+	uint64_t                 polynomial;
+	const void               *table;
+};
+
+/*
+ * This structure is materialized in each shared object file that defines
+ * a CRC.  Therefore, instead of checking the structure address, the
+ * `wrap_struct_name` field is simply checked.  This is to avoid link errors
+ * at build time.
+ */
+static const rb_data_type_t crc_model_const_type = {
+	"digest-crc:model-const",
+	{
+		NULL,
+		NULL,
+		NULL,
+		NULL
+	}
+};
+
+static inline void ruby_digest_crc_model_implant(VALUE crc_class, const struct crc_model *model)
+{
+	ID id_model = rb_intern("digest-crc-model");
+	VALUE obj = rb_data_typed_object_wrap(rb_cObject, (void *)(uintptr_t)model, &crc_model_const_type);
+	rb_obj_freeze(obj);
+	rb_ivar_set(crc_class, id_model, obj);
+}
+
+#if !defined(HAVE_RB_EXT_RACTOR_SAFE)
+static inline void ruby_digest_crc_ensure_ractor_main(const char *mesg)
+{
+	(void)mesg;
```

Comment:
Not sure what the purpose is of this statement? Seems to be a no-op?

Reply:
This is used to suppress `cc -Wextra` warnings about unused variables.